logs/replace.go

```diff
@@ -17,7 +18,7 @@ func ReplaceSecrets(logLines []state.LogLine, filterLogSecret []state.LogSecretK
 		}
 	}
 	for idx, logLine := range logLines {
-		if filterUnidentified && logLines[idx].Classification == 0 {
+		if filterUnidentified && logLines[idx].Classification == 0 && logLines[idx].ParentUUID == uuid.Nil {
```

Comment:
I'm not sure if this is actually sufficient, since this would mean that child log lines of unidentified parent log lines would never be redacted, right? Is there a better way to handle this?

Reply:
Is it even possible for an unidentified log line to have children? I'd have assumed the parent-child association relies on being able to parse the log line.

Reply:
I think so: my reading of [the logic to assign parents to log lines](https://github.com/pganalyze/collector/blob/main/logs/analyze.go#L2212-L2246) is that it doesn't care what the parent log line actually is. That code is pretty intricate, though, so I may be missing something.

Reply:
Per separate discussion, this may still be an issue, but it's an edge case and we can fix that in a follow-up patch.